stacks-signer/src/client/stacks_client.rs

```diff
@@ -302,29 +440,68 @@ impl StacksClient {
     }
 
     /// Sends a transaction to the stacks node for a modifying contract call
-    #[allow(dead_code)]
-    fn transaction_contract_call(
+    pub fn cast_vote_for_aggregate_public_key(
         &self,
-        contract_addr: &StacksAddress,
-        contract_name: ContractName,
-        function_name: ClarityName,
-        function_args: &[ClarityValue],
+        point: Point,
+        round: u64,
     ) -> Result<Txid, ClientError> {
-        debug!("Making a contract call to {contract_addr}.{contract_name}...");
+        debug!("Casting vote for aggregate public key to the mempool...");
+        let signed_tx = self.build_vote_for_aggregate_public_key(point, round)?;
+        self.submit_tx(&signed_tx)
+    }
+
+    /// Helper function to create a stacks transaction for a modifying contract call
+    pub fn build_vote_for_aggregate_public_key(
+        &self,
+        point: Point,
+        round: u64,
+    ) -> Result<StacksTransaction, ClientError> {
+        debug!("Building vote-for-aggregate-public-key transaction...");
+        let signer_index = 0; // TODO retreieve the index from the stacks node
```

Comment:
I think you have a `get_signer_index()` you can use here?

Reply:
Yep! It is is currently kinda broken though...need to update afer yor changes go through for updating stackerdb I think.